### PR TITLE
Initial stats API

### PR DESCRIPTION
**Pull request by hugoArregui**
Related to https://github.com/pion/webrtc/pull/721

### CODE REVIEW

**Comment by hugoArregui** [comment]
@maxhawkins @Sean-Der will you guys be ok by merging this as it is and create an issue to track the missing stats? (the ones commented in agent.go). I don't want to create a huge PR nor to have to a PR open for long and have to rebase all the time.

**Comment by enobufs** [comment]
Can't wait to see this landing!! stats, stats, Stats!

**Comment by hugoArregui** [comment]
@enobufs I know you wanted to use this to get candidate pairs, will your use case be covered by the info I added in this PR?

File: util.go [hunk 1]
```diff
@@ -73,3 +74,7 @@ func addrEqual(a, b net.Addr) bool {
 
 	return aType == bType && aIP.Equal(bIP) && aPort == bPort
 }
+
+func rndCandidateID() string {
+	return fmt.Sprintf("candidate:%d", time.Now().UnixNano())
```
Comment (maxhawkins) [inline util.go]: What if this is called concurrently? Maybe we should use a UUID?

**Comment by maxhawkins** [comment]
Have you given any thought to how this would be surfaced in PeerConnection? We should probably have a GetStats() method to match the browser implementation.

**Comment by enobufs** [comment]
> @enobufs I know you wanted to use this to get candidate pairs, will your use case be covered by the info I added in this PR?

It looks exactly what I would like to have for my diagnostic purpose!

File: util.go (hunk 1 above)
Comment (hugoArregui) [reply util.go]: yes, that would work too, I'm using nano so I think a clash is very unlikely, but we could use a uuid library too

**Comment by hugoArregui** [comment]
> Have you given any thought to how this would be surfaced in PeerConnection? We should probably have a GetStats() method to match the browser implementation.

Yes, I implemented a GetStats as well https://github.com/pion/webrtc/pull/721/files#diff-d2ee665bb8c62c914fee5e9c743be25cR1836

File: util.go (hunk 1 above)
Comment (Sean-Der) [reply util.go]: @hugoArregui we could use https://github.com/pion/ice/blob/master/mdns.go#L31

If you want to move it into `util.go` and have the caller append `.local`

Comment (hugoArregui) [reply util.go]: good call! done

**Pull request merged**